### PR TITLE
Add tests for rulesoftype

### PR DESCRIPTION
**Pull request by mhauru**
@pwochner

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -65,6 +65,36 @@ using Test
...
+                            RuleNode(5, [RuleNode(7)]),
+                            RuleNode(6, [RuleNode(9, [RuleNode(10)])])
+                        ]
+                    )
+                ]
+            )
+            @test rulesoftype(rulenode, Set((11, 12))) == Set{Int}()
+            arg = Set((1, 3, 7, 9, 10, 15, 23))
+            expected = Set((1, 3, 7, 9, 10))
+            @test rulesoftype(rulenode, arg) == expected
```
Comment (ReubenJ) [inline test/runtests.jl:92]:
```suggestion
            ruleset = Set((1, 3, 7, 9, 10, 15, 23))
            expected = Set((1, 3, 7, 9, 10))
            @test rulesoftype(rulenode, ruleset) == expected
```

File: test/runtests.jl [hunk 2]
```diff
@@ -65,6 +65,36 @@ using Test
...
+            @test rulesoftype(rulenode, Set(1)) == Set(1)
+            @test rulesoftype(rulenode, Set{Int}()) == Set{Int}()
```
Comment (ReubenJ) [inline test/runtests.jl:94]:
```suggestion
            @test isempty(rulesoftype(rulenode, Set((11, 12))))
            arg = Set((1, 3, 7, 9, 10, 15, 23))
            expected = Set((1, 3, 7, 9, 10))
            @test rulesoftype(rulenode, arg) == expected
            @test rulesoftype(rulenode, Set(1)) == Set(1)
            @test isempty(rulesoftype(rulenode, Set{Int}()))
```

**Review by ReubenJ — CHANGES_REQUESTED** [review]
Thanks! Just a couple of small suggestions.

File: test/runtests.jl (hunk 2 above)
Comment (mhauru) [reply test/runtests.jl]: Done.

**Review by ReubenJ — APPROVED** [review]
Great. Thanks!

**Pull request merged**